fsspec/implementations/tar.py

```diff
@@ -88,22 +88,25 @@ def _index(self):
         out = {}
         for ti in self.tar:
             info = ti.get_info()
-            info["type"] = typemap[info["type"]]
+            info["type"] = typemap.get(info["type"], "file")
             name = ti.get_info()["name"].rstrip("/")
             out[name] = (info, ti.offset_data)
 
         self.index = out
         # TODO: save index to self.index_store here, if set
 
     def _get_dirs(self):
-
         if self.dir_cache is not None:
             return
 
-        self.dir_cache = {}
+        # This enables ls to get directories as children as well as files
+        self.dir_cache = {
+            dirname + "/": {"name": dirname + "/", "size": 0, "type": "directory"}
```

Comment:
I would discourage the use of the "/" suffix, since other implementations don't do this. We already say it's a directory, so no need to indicate it again.

Reply:
@martindurant interesting, I actually cloned this from the ZipFileSystem's implementation: https://github.com/aramesh117/filesystem_spec/blob/f7237f8c3472db9a795213e080d709da994b4c57/fsspec/implementations/zip.py#L69. I can certainly remove it, but just wondering why the difference in behavior.

Reply:
Done.

Reply:
Seems like removing the '/' suffix breaks the info test. The expected value includes a / suffix. What would you recommend?

Reply:
OK, well I suppose put it back, and we can consider a separate PR where we bring TAR and other archive formats into line.